tree-sitter-tips.md

```diff
@@ -0,0 +1,79 @@
+
+### :bulb: What does a "good" parse tree look like?
+
+Here are some attributes of a "pretty" tree:
+
+:white_check_mark: Not overly verbose.
+
+:white_check_mark: Everything is outlined clearly.
+
+:white_check_mark: Communicates good information needed further along in the pipeline.
+
+![image](https://user-images.githubusercontent.com/875834/34972577-41bc12ac-fa37-11e7-9a12-a05d18b6f6e4.png)
+
+_Example of a good tree (WIP -- better examples welcome!):_
+
+```
+===
+module
+===
+
+module com.foo { }
+open module com.foo { }
+
+---
+
+(program
+  (module_declaration (module_identifier (identifier) (identifier)))
+  (module_declaration (module_identifier (identifier) (identifier))))
+```
+
+### :bulb: Improving a parse tree by making it more compact
+Here are some things that might help:
+
+**1. Aliasing** - by aliasing different things, you can maintain compact tree structure while marking differences in your grammar. _(Need better description)_
+
+**2. Making things hidden** - by putting an underscore before a rule (`_rule` vs `rule`), it will allow you to omit displaying it in the tree.
+
+### :bulb: Testing in the wild to help prioritize what to support next.
+
+Once you have developed a significant portion of the grammar, find a file from an active open source repository that follows the typical structure of the language and see how much of it can be successfully parsed by your system. Is it 40%? 60%? This can be a good way to benchmark how practical your grammar is in terms of popular usage of your language. It can also be an effective way to identify what to support and prioritize what's next.
+
+### :bulb: Knowing when to stop development
+Most languages have a long-tail of features that don't get utilized frequently. It is reasonable to prioritize developing 80% of a language and only supporting commonly used elements.
+
+### :bulb: Handling conflicts
+
+Conflicts may arise due to ambiguities in the grammar. This is when the parser can not decide what the next symbol in an input stream should be because there are multiple ways to parse some strings. There are three ways to handle conflict:
```

Comment:
I think this is the part of grammar development that still often confuses me the most: what do you do when there are conflicts? tree-sitter-cli does give some hints for the specific situation - maybe it would help to walk through a concrete example of the kind of conflicts you might run into?